src/agent/configuration_parser/tests/configuration_parser_test.cpp

```diff
@@ -1,54 +1,144 @@
 #include "configuration_parser.hpp"
 #include <map>
+#include <gtest/gtest.h>
 
 
-// File added as a test to carry out manual development tests
-// To run this test there must be a file named wazuh.conf in the execution directory.
-
-int main()
-{
-    configuration::ConfigurationParser *parser = new configuration::ConfigurationParser;
-
-    // [agent]
-    // manager_ip = "192.168.0.11"
-    // string_conf = "string"
-    std::cout << "###### Agent - manager_ip ######" << "\n\n";
-    const auto t = parser->GetConfig<std::string>("agent", "manager_ip");
-    assert(t == "192.168.0.11");
-    std::cout << t << "\n\n";
-
-    // [agent_array]
-    // array_manager_ip = ["192.168.0.0", "192.168.0.1"]
-    // string_conf = "string"
-    std::cout << "###### Agent - array_manager_ip ######" << "\n\n";
-    const auto tt = parser->GetConfig<std::vector<std::string>>("agent_array", "array_manager_ip");
-    assert(tt[0] == "192.168.0.0");
-    assert(tt[1] == "192.168.0.1");
-    std::cout << tt[0] << "\n\n";
-    std::cout << tt[1] << "\n\n";
-
-    // [agent]
-    // manager_ip = "192.168.0.11"
-    // string_conf = "string"
-    std::cout << "###### Agent - map ######" << "\n\n";
-    auto agent_map = parser->GetConfig<std::map<std::string, std::string>>("agent");
-    for(const auto elem : agent_map)
-    {
-        std::cout << elem.first << ": " << elem.second << "\n";
-    }
-
-    // [bad_map]
-    // string = "str_conf"
-    // int = 1
-    std::cout << "###### Bad mapping ######" << "\n\n";
-    try
-    {
-        auto agent_map = parser->GetConfig<std::map<std::string, std::string>>("bad_map");
-    }
-    catch(const std::exception& e)
-    {
-        std::cout << "Exception caught" << '\n';
-    }
-
-    return 0;
+using namespace configuration;
```

Comment:
It would be better to introduce the tests from the beginning instead of creating the `main()` function.